File: Ejercicios Corredor/Trabajo Corredor/Booleano.go
package main
import "fmt"

type Booleano struct {
	activo     bool
	encendido  bool
	aprobado   bool
	disponible bool
	visible    bool
}

func (b Booleano) EsVerdadero(valor bool) bool { return valor }
func (b Booleano) Negacion(valor bool) bool { return !valor }
func (b Booleano) AmbosVerdaderos(a, c bool) bool { return a && c }
func (b Booleano) AlgunoVerdadero(a, c bool) bool { return a || c }
func (b Booleano) Iguales(a, c bool) bool { return a == c }

func main() {
	fmt.Println(Booleano{}.AmbosVerdaderos(true, false))
}

File: Ejercicios Corredor/Trabajo Corredor/Booleano.cs
public class Booleano {
    public bool Activo = true;
    public bool Encendido = false;
    public bool Aprobado = true;
    public bool Disponible = false;
    public bool Visible = true;

    public bool EsVerdadero(bool valor) => valor;
    public bool Negacion(bool valor) => !valor;
    public bool AmbosVerdaderos(bool a, bool b) => a && b;
    public bool AlgunoVerdadero(bool a, bool b) => a || b;
    public bool Iguales(bool a, bool b) => a == b;
}
